src/main/java/org/jsoup/parser/Tokeniser.java

```diff
@@ -174,36 +202,42 @@ void advanceTransition(TokeniserState newState) {
         reader.advance();
     }
 
-    final private int[] codepointHolder = new int[1]; // holder to not have to keep creating arrays
+    // holder to not have to keep creating arrays
+    final private int[] codepointHolder = new int[1];
+
     final private int[] multipointHolder = new int[2];
-    @Nullable int[] consumeCharacterReference(@Nullable Character additionalAllowedCharacter, boolean inAttribute) {
+
+    @Nullable
+    int[] consumeCharacterReference(@Nullable Character additionalAllowedCharacter, boolean inAttribute) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 27. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.